docs/decisions/0055-declarative-integrations-with-fakes.md

```
@@ -0,0 +1,600 @@
+test_webhookDispatch = do
+  TestIntegration.simulate @StripeWebhook syntheticEvent
+  ...
+```
+
+Compile-time exclusion from production module paths is a stronger guarantee than a runtime gate. Nick overrides `runFake` only when the inbound needs its own autonomous behaviour in smoke tests (e.g. a fake file watcher that periodically emits synthetic files).
+
+### 3. Selection via `Application.run` — CLI Flag Gated by Environment Variable
+
+Selection happens in the default CLI surface of `Application.run`. No Haskell source change is required to flip from real to fake. The CLI flag alone is insufficient: `--integrations=fake` and `--integrations=hybrid --fake=NAME` are **only honoured** when the environment variable `NEOHASKELL_ALLOW_FAKE_INTEGRATIONS=1` is present in the process environment. Without the env var, `Application.run` rejects the flag at startup with an error and exits non-zero. The error text must include (a) the offending flag, (b) the exact env-var name and required value `NEOHASKELL_ALLOW_FAKE_INTEGRATIONS=1`, and (c) a pointer to the test-wiring documentation. For example:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)